packages/story-editor/src/components/floatingMenu/layer.js

```diff
@@ -50,11 +55,35 @@ function FloatingMenuLayer() {
       workspaceHeight,
     })
   );
+  // We need to know if anything (non-bg) is selected, if more than one element is selected,
+  // the list of different element types selected as well as a unique identifier
+  // for the combined selection (in order to undismiss in case of change of seletion)
+  const {
+    hasSelection,
+    hasMultiSelection,
+    selectionTypes,
+    selectionIdentifier,
+  } = useStory(({ state: { selectedElements } }) => ({
+    hasSelection:
+      selectedElements.filter(({ isBackground }) => !isBackground).length > 0,
+    hasMultiSelection: selectedElements.length > 1,
+    selectionTypes: [
+      ...new Set(selectedElements.map(({ type }) => type)),
+    ].sort(),
+    selectionIdentifier: selectedElements
+      .map(({ id }) => id)
+      .sort()
+      .join('#'),
```

Comment:
Curious why is `#` needed, for debugging?

Reply:
Do we even need to sort this? (You can't just swap elements in place 🤔 )

![image](https://user-images.githubusercontent.com/21036224/153224164-229eccdc-3da4-434f-bbfd-c55613cb877e.png)

Reply:
@miina, this is all assuming that id's can be anything, so they could potentially collide if e.g. you come from a selection of '123' and '456' joined without separator to a selection of just '123456'. But if we assume that id's are always same length, then there isn't an issue here.

@merapi, good point. I wasn't sure whether lasso selection always selected elements in the same order, so you might make two lasso selections, that selected the same elements, but in different order. If that's not an issue, than we don't need to sort. Also, does sort of 5 strings really take more than a second? WTF?

Reply:
Changed both parts in 125649ca00.

Reply:
Nah, I always do those kinds of measurements in a 1k-1kk loop, but on a potato laptop and many elements, it can make a difference.

Lasso: that second lasso will/should reset the dismiss flag so it's not an issue(?)